File: lua/cyboquatic/workload_assess.lua
-- File: lua/cyboquatic/workload_assess.lua
local gate = require("cyboquatic.workload_gate")

local names = {
  "flow_m3_s", "lift_m", "efficiency", "runtime_s", "voltage_drop_v",
  "renewable_fraction", "embodied_carbon_g_per_j", "biodiversity_risk"
}
local sample = {
  node_id = "phoenix-canal-pump-01",
  owner_did = "bostrom18sd2ujv24ual9c9pshtxys6j8knh6xaead9ye7",
  canal_node = "phoenix-canal-pump-01",
  K = 0.94, E = 0.91, R = 0.12, fog_confidence = 0.90
}

for index, name in ipairs(names) do
  local value = tonumber(arg[index])
  assert(value, "expected eight numeric telemetry values")
  sample[name] = value
end

local result = gate.evaluate(sample)
assert(result.frame, result.reason or "assessment failed")
print("energyreqJ=" .. result.frame.energyreqJ)
print("deltaVt=" .. result.frame.deltaVt)
print("knowledge_factor=" .. result.frame.knowledgeFactor)
print("eco_impact_value=" .. result.frame.ecoImpactValue)
print("accepted=" .. (result.accepted and "1" or "0"))

File: crates/round-trip-verify/src/bin/workload_assess.rs
// File: cyboquatic-core/src/bin/workload_assess.rs
use std::{env, process};

use cyboquatic_core::{assess_workload, WorkloadAssessment, WorkloadInput};

fn parse() -> WorkloadInput {
    let values: Vec<f64> = env::args()
        .skip(1)
        .map(|value| value.parse::<f64>())
        .collect::<Result<_, _>>()
        .unwrap_or_else(|_| {
            eprintln!("expected eight finite telemetry values");
            process::exit(64);
        });
    if values.len() != 8 {
        eprintln!("expected: flow lift efficiency runtime voltage renewable carbon biodiversity");
        process::exit(64);
    }
    WorkloadInput {
        flow_m3_s: values[0],
        lift_m: values[1],
        efficiency: values[2],
        runtime_s: values[3],
        voltage_drop_v: values[4],
        renewable_fraction: values[5],
        embodied_carbon_g_per_j: values[6],
        biodiversity_risk: values[7],
    }
}

fn main() {
    let input = parse();
    let mut output = WorkloadAssessment::default();
    if assess_workload(&input, &mut output) != 0 {
        eprintln!("invalid telemetry");
        process::exit(65);
    }
    println!("energyreqJ={}", output.energyreq_j);
    println!("deltaVt={}", output.delta_vt);
    println!("knowledge_factor={}", output.knowledge_factor);
    println!("eco_impact_value={}", output.eco_impact_value);
    println!("accepted={}", output.accepted);
}
